helios.js

```diff
@@ -16704,7 +16707,7 @@ class ValidatorHashType extends HashType {
 		switch (name.value) {
 			case "CURRENT":
 				if (this.macrosAllowed) {
-					if (this.#purpose == ScriptPurpose.Spending || this.#purpose == ScriptPurpose.Testing) {
+					if (this.#purpose == ScriptPurpose.Spending || this.#purpose == ScriptPurpose.Staking || this.#purpose == ScriptPurpose.Testing) {
```

Comment:
I think `ValidatorHash::CURRENT` only makes sense in a spending script.

For staking scripts `CURRENT` would probably have to be part of `StakingCredential` instead.

Reply:
(the other stuff is great btw, thanks!)

Reply:
Yes. I think about `StakeValidatorHash` but it is just a wrapper in `Plutus`, so `ValidatorHash` is fine